src/coreComponents/constitutive/fluid/PVTFunctions/BrineCO2DensityFunction.cpp

```diff
@@ -26,69 +29,57 @@ using namespace stringutilities;
 namespace PVTProps
 {
 
-BrineCO2DensityFunction::BrineCO2DensityFunction( string_array const & inputPara,
-                                                  string_array const & componentNames,
-                                                  real64_array const & componentMolarWeight ):
-  PVTFunction( inputPara[1], componentNames, componentMolarWeight )
+BrineCO2Density::BrineCO2Density( array1d< string > const & inputPara,
+                                  array1d< string > const & componentNames,
+                                  array1d< real64 > const & componentMolarWeight ):
+  PVTFunctionBase( inputPara[1],
+                   componentNames,
+                   componentMolarWeight )
 {
-  bool notFound = 1;
-
+  bool notFound = true;
   for( localIndex i = 0; i < componentNames.size(); ++i )
   {
-
-    if( streq( componentNames[i], "CO2" ) || streq( componentNames[i], "co2" ))
+    if( componentNames[i] == "CO2" || componentNames[i] == "co2" )
     {
       m_CO2Index = i;
-      notFound = 0;
+      notFound = false;
       break;
     }
-
   }
-
   GEOSX_ERROR_IF( notFound, "Component CO2 is not found!" );
 
-  notFound = 1;
-
+  notFound = true;
   for( localIndex i = 0; i < componentNames.size(); ++i )
   {
-
-    if( streq( componentNames[i], "Water" ) || streq( componentNames[i], "water" ))
+    if( componentNames[i] == "Water" || componentNames[i] == "water" )
     {
       m_waterIndex = i;
-      notFound = 0;
+      notFound = false;
       break;
     }
-
   }
-
   GEOSX_ERROR_IF( notFound, "Component Water/Brine is not found!" );
 
-
   makeTable( inputPara );
 }
 
-void BrineCO2DensityFunction::makeTable( string_array const & inputPara )
+void BrineCO2Density::makeTable( array1d< string > const & inputPara )
 {
-  real64_array pressures;
-  real64_array temperatures;
-
-  real64 PStart, PEnd, dP;
-  real64 TStart, TEnd, dT;
-  real64 P, T, m;
-
-  dT = -1.0;
-  dP = -1.0;
-  TStart = -1.0;
-  TEnd = -1.0;
-  PStart = -1.0;
-  PEnd = -1.0;
+  array1d< array1d< real64 > > coordinates;
+  coordinates.resize( 2 );
```

Comment:
Would it make sense to build some `TableCoordinates class` that would whether inherit from `array1d< array1d< real64 > >` (🙄) or has a `array1d< array1d< real64 > >` attribute to bring more readability to your code.

E.g. the `coordinates.resize( 2 )` could be imposed in the constructor (instead of being done 4 different times in your code).
And we could replace `coordinates[0].size()` by `coordinates.nPressures()`, or `coordinates[0].emplace_back( P )` by `coordinates.appendPressure( P )` and `coordinates[1][j]` by `coordinates.getPressure( j )`. 
The indices `0` and `1` would be hidden when unnecessary (maybe always?) and we could still get no call cost by imposing `inline` (IIUC).

Reply:
I added the class that you suggested here: https://github.com/GEOSX/GEOSX/blob/0c489b262db7da805c43439da199eaa7ef7aca63/src/coreComponents/constitutive/fluid/PVTFunctions/PVTFunctionHelpers.hpp#L54
so there are no `0` and `1` indices any more in the code you referenced and it is nicer.